src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs

```diff
@@ -520,6 +534,8 @@ public async Task<HttpResponseMessage> SendAsyncCore(HttpRequestMessage request,
                 var response = new HttpResponseMessage() { RequestMessage = request, Content = new HttpConnectionResponseContent() };
                 ParseStatusLine(await ReadNextResponseHeaderLineAsync(async).ConfigureAwait(false), response);
 
+                if (HttpTelemetry.Log.IsEnabled()) HttpTelemetry.Log.ResponseHeadersBegin();
```

Comment:
This is oddly specific event. However, should it rather be nearer line 575: https://github.com/dotnet/runtime/pull/40838/files#diff-75931c95adb5205b557555a5cd4f966eR575 ?

Reply:
The idea behind this event is that there is a timestamp around the time we start receiving some request-related response from the server. This way, latency issues caused by the server's slow initial response can be observed.